docs/design/datacontracts/data_descriptor.md

```diff
@@ -92,6 +93,15 @@ The value must be an integral constant within the range of its type.  Signed val
 natural encoding.  Pointer values need not be aligned and need not point to addressable target
 memory.
 
+## Sub-descriptor descriptors
+
+Each sub-descriptor descriptor is effectively a global with a type of `pointer`. They will consist of:
+* a name
+* a pointer value
+
+If these values are non-null, the pointer represents another JSON data descriptor with the specification described in this document.
+
+When parsing a data descriptor with sub-descriptors each sub-descriptor should be parsed then its type, global, and contract values should be merged in. If any conflicts arise when merging in sub-descriptor data, this is an error and behavior is undefined.
```

Comment:
> If any conflicts arise when merging in sub-descriptor data, this is an error and behavior is undefined.

This design means that the components involved need to be aware of each other to avoid conflicts. Just pointing it out.